File: Bubble-Sort/python/leetcode_lc_45.py
nums = [3,30,34,5,9]

class Solution:
    def minNumber(self, nums) -> str:
        nums = [str(v) for v in nums]
        for i in range(0,len(nums)):
            for j in range(1,len(nums)-i):
                if nums[j-1]+nums[j] > nums[j]+nums[j-1]:
                    nums[j-1],nums[j] = nums[j],nums[j-1]

        return "".join(nums)
    
solution = Solution()

print(solution.minNumber(nums))

File: Bubble-Sort/js/leetcode_lc_45.js
const nums = [3,30,34,5,9]

/**
 * @param {number[]} nums
 * @return {string}
 */
var minNumber = function(nums) {
    for(let i = 0 ; i < nums.length ; i++){
        for(let j = 1 ; j < nums.length - i ; j++){
            if(nums[j-1].toString() + nums[j].toString() > nums[j].toString() + nums[j-1].toString()){
                const tmp = nums[j-1]
                nums[j-1] = nums[j]
                nums[j] = tmp
            }
        }
    }
    return nums.join('')
};

console.log(minNumber(nums))